### PR TITLE
storagecluster: remove the deviceType from validateStorageDeviceSets

### PR DESCRIPTION
**Pull request by crombus**
Remove the metadata setting from storage cluster as it is benefiting nothing.
just check for invalid deviceType and throw error.

Signed-off-by: crombus <pkundra@redhat.com>

### CODE REVIEW

**Review by obnoxxx — CHANGES_REQUESTED** [review]
What's the purpose of this PR? Why is it needed? Please elaborate in commit message / PR description.

File: pkg/controller/storagecluster/reconcile.go [hunk 1]
```diff
@@ -425,15 +425,15 @@ func (r *ReconcileStorageCluster) validateStorageDeviceSets(sc *ocsv1.StorageClu
...
 			}
 		}
+
 		if ds.DeviceType != "" {
 			if (DeviceTypeSSD == strings.ToLower(ds.DeviceType)) || (DeviceTypeHDD == strings.ToLower(ds.DeviceType)) || (DeviceTypeNVMe == strings.ToLower(ds.DeviceType)) {
-				metav1.SetMetaDataAnnotation(&sc.ObjectMeta, "crushDeviceClass", ds.DeviceType)
+				// Leaving this empty because it is just to validate the DeviceType
 			} else {
 				return fmt.Errorf("failed to validate DeviceType %q: no Device of this type", ds.DeviceType)
 			}
```
Comment (iamniting) [inline pkg/controller/storagecluster/reconcile.go:434]: You can rephrase it to 

if (DeviceTypeSSD != strings.ToLower(ds.DeviceType)) && (DeviceTypeHDD != strings.ToLower(ds.DeviceType)) && (DeviceTypeNVMe != strings.ToLower(ds.DeviceType)) {
return fmt.Errorf("failed to validate DeviceType %q: no Device of this type", ds.DeviceType)
}

**Review by iamniting — CHANGES_REQUESTED** [review]

File: pkg/controller/storagecluster/reconcile.go (hunk 1 above)
Comment (crombus) [reply pkg/controller/storagecluster/reconcile.go]: done

File: pkg/controller/storagecluster/reconcile.go [hunk 2]
```diff
@@ -425,15 +425,13 @@ func (r *ReconcileStorageCluster) validateStorageDeviceSets(sc *ocsv1.StorageClu
 				return fmt.Errorf("failed to validate StorageDeviceSet %d: no StorageClass specified for walPVCTemplate", i)
 			}
 		}
+
 		if ds.DeviceType != "" {
-			if (DeviceTypeSSD == strings.ToLower(ds.DeviceType)) || (DeviceTypeHDD == strings.ToLower(ds.DeviceType)) || (DeviceTypeNVMe == strings.ToLower(ds.DeviceType)) {
-				metav1.SetMetaDataAnnotation(&sc.ObjectMeta, "crushDeviceClass", ds.DeviceType)
-			} else {
+			if (DeviceTypeSSD != strings.ToLower(ds.DeviceType)) || (DeviceTypeHDD != strings.ToLower(ds.DeviceType)) || (DeviceTypeNVMe != strings.ToLower(ds.DeviceType)) {
```
Comment (iamniting) [inline pkg/controller/storagecluster/reconcile.go:430]: The conditions should be && instead of ||

**Review by iamniting — CHANGES_REQUESTED** [review]

File: pkg/controller/storagecluster/reconcile.go (hunk 2 above)
Comment (crombus) [reply pkg/controller/storagecluster/reconcile.go]: I was confused about it but after checking through all conditions && is the appropriate operator . Adding || will fail the condition. Done

**Review by iamniting — APPROVED** [review]

**Review by obnoxxx — APPROVED** [review]
/lgtm
